### DISCUSSION TITLE
ContainerAccess allows minecraft destruction

### DISCUSSION THREAD

**Discussion by Jerethkhan** [GriefPrevention/GriefPrevention#640] [General]
### What happened:
When /containeraccess perms are given to a play, they allow that player to destroy hopper/furnace/chest minecarts.

### What you expected:
Container access not allowing the destruction of minecart containers.

### Steps to reproduce:
1. Give someone only container trust.
2. Place a chest minecart.
3. Watch them destroy it.

### Server and GriefPrevention version:
PaperMC build 645 (1.13.2)
GriefPrevention 16.11.6

**Comment by RoboMWM** [comment]
Vehicle destruction is part of containertrust iirc.

**Comment by Jerethkhan** [comment]
> 
> 
> Vehicle destruction is part of containertrust iirc.

Obviously. If it is intended behavior, it should not be. It allows griefing of containers and machines such as auto-sorters, auto-smelters, and farm collection systems.

**Comment by RoboMWM** [comment]
> such as auto-sorters, auto-smelters, and farm collection systems.

Which all consist of containers...?

**Comment by Jikoo** [comment]
OP is talking about non-ridable minecart varieties, which are almost exclusively used in automation.
For example, one of my filter designs that accepts items at double the rate of a standard hopper-based item filter:
![image](https://user-images.githubusercontent.com/1353270/66714490-7de36a80-ed85-11e9-9680-e6c998f634a0.png)

It's certainly possible to do all the things OP is describing without minecarts or with properly laid out machines and subclaims, but it would be a nice QoL change for simplicity.

**Comment by RoboMWM** [comment]
Ok, but I'm not sure what containertrust has to do with this..?

**Comment by Jikoo** [comment]
Currently destroying them requires container trust, not build trust. OP is arguing that minecart variants aside from the ones that can actually be ridden are only likely to be used inside complex devices and should require build trust to destroy, similar to armor stands and item frames.

**Comment by lmichaelis** [comment]
I do agree with @Jerethkhan here. This shouldn't be an intended mechanic since minecarts are used for trapping villagers for public trading too. If I wanted for someone to be able to destroy things I would give them full permissions instead of container trust. The relevant code here is in [EntityEventHandler.java](https://github.com/TechFortress/GriefPrevention/blob/8c7abf9fd56f7dfdf496cb038221f405d0b29bd2/src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java#L1331)

**Comment by RoboMWM** [comment]
There's a lot of discussion on primarily two sides, one want more restrictive permissions (such as this) and the other wanting less restrictive (such as https://korobi.vq.lc/network/spigot/channel/griefprevention/logs/2020/10/21/#L10).

If you can craft a proposal that fits with the original design/intent of the current permissions system I'll consider it. The only entities I know of that are protected with build trust are all "building/decor-related" items such as itemframes, paintings, armorstands, etc. Animals are "moveable" entities and are protected with containertrust. Tameable animals are their own thing that differs if the world is set as pvp or not. You may also be able to make a claim based on villagers being protected perhaps... but at best interaction with those could be considered to be moved to containertrust instead.

**Reply by parmaia** [reply]
Sorry to necro this issue, but what about allowing a configuration to decide if allow break non-rideable minecarts with containertrust or buildtrust.
